cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -0,0 +1,98 @@
+package com.sequenceiq.cloudbreak.cloud.azure;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.microsoft.azure.management.compute.implementation.DiskEncryptionSetInner;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.sequenceiq.cloudbreak.cloud.EncryptionResources;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClient;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClientService;
+import com.sequenceiq.cloudbreak.cloud.model.Platform;
+import com.sequenceiq.cloudbreak.cloud.model.Variant;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.CreatedDiskEncryptionSet;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.DiskEncryptionSetCreationRequest;
+
+@Service
+public class AzureEncryptionResources implements EncryptionResources {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureEncryptionResources.class);
+
+    private static final Pattern ENCRYPTION_KEY_URL_PATTERN = Pattern.compile("https://([^.]+)\\\\.vault.*");
+
+    @Inject
+    private AzureClientService azureClientService;
+
+    @Inject
+    private AzureUtils azureUtils;
+
+    @Override
+    public Platform platform() {
+        return AzureConstants.PLATFORM;
+    }
+
+    @Override
+    public Variant variant() {
+        return AzureConstants.VARIANT;
+    }
+
+    @Override
+    public CreatedDiskEncryptionSet createDiskEncryptionSet(DiskEncryptionSetCreationRequest diskEncryptionSetCreationRequest) {
+        AzureClient azureClient = azureClientService.getClient(diskEncryptionSetCreationRequest.getCloudCredential());
+        String encryptionKeyUrl = diskEncryptionSetCreationRequest.getEncryptionKeyUrl();
+        String envName = diskEncryptionSetCreationRequest.getEnvName();
+        String location = diskEncryptionSetCreationRequest.getRegion().getRegionName();
+        Map<String, String> tags = diskEncryptionSetCreationRequest.getTags();
+        String vaultName = ENCRYPTION_KEY_URL_PATTERN.matcher(encryptionKeyUrl).group(1);
+
+        String desResourceGroupName = null;
+        String vaultResourceGroupName = null;
+        if (diskEncryptionSetCreationRequest.isSingleResourceGroup()) {
+            desResourceGroupName = diskEncryptionSetCreationRequest.getResourceGroup();
+            vaultResourceGroupName = desResourceGroupName;
+        } else {
+            desResourceGroupName = (String) createResourceGroup(azureClient, envName, location, tags);
+            vaultResourceGroupName = azureClient.getVaultResourceGroup(vaultName);
```

Comment:
As mentioned elsewhere, the Key Vault RG / Resource ID always has to be determined (or supplied by the user via the environment create request), irrespective of Single-RG vs Multi-RG. Single-RG only means that CB will create all the resources in a dedicated pre-existing RG, but it does not say anything about the location of other resources provided & managed by the user (e.g. Storage Account, MSI, Key Vault).

Therefore, the statement of line 61 can be moved here, and line 58 shall be deleted.